modules/apps/captcha/captcha-api/src/main/java/com/liferay/captcha/util/CaptchaUtil.java

```diff
@@ -24,49 +28,116 @@
 public class CaptchaUtil {
 
 	public static void check(HttpServletRequest httpServletRequest)
-		throws CaptchaException {
+		throws CaptchaConfigurationException, CaptchaException {
 
-		getCaptcha().check(httpServletRequest);
+		getCaptcha(
+			httpServletRequest
+		).check(
+			httpServletRequest
+		);
 	}
 
 	public static void check(PortletRequest portletRequest)
-		throws CaptchaException {
+		throws CaptchaConfigurationException, CaptchaException {
 
-		getCaptcha().check(portletRequest);
+		getCaptcha(
+			portletRequest
+		).check(
+			portletRequest
+		);
 	}
 
-	public static void enforceCaptcha(HttpServletRequest httpServletRequest) {
-		getCaptcha().enforceCaptcha(httpServletRequest);
+	public static void enforceCaptcha(HttpServletRequest httpServletRequest)
+		throws CaptchaConfigurationException {
+
+		getCaptcha(
+			httpServletRequest
+		).enforceCaptcha(
+			httpServletRequest
+		);
 	}
 
-	public static void enforceCaptcha(PortletRequest portletRequest) {
-		getCaptcha().enforceCaptcha(portletRequest);
+	public static void enforceCaptcha(PortletRequest portletRequest)
+		throws CaptchaConfigurationException {
+
+		getCaptcha(
+			portletRequest
+		).enforceCaptcha(
+			portletRequest
+		);
 	}
 
-	public static Captcha getCaptcha() {
-		CaptchaProvider captchaProvider = _captchaProviderSnapshot.get();
+	public static Captcha getCaptcha(HttpServletRequest httpServletRequest)
+		throws CaptchaConfigurationException {
+
+		try {
+			CaptchaConfiguration captchaConfiguration =
```

Comment:
Hey @ChrisKian, move the logic to retrieve the `CaptchaConfiguration` to the CaptchaProvider so we can avoid the changes on the CaptchaUtil class. Also, try to use the `CompanyThreadLocal#getCompanyId` instead of the `PortalUtil#getCompanyId`.

Thanks